lib/fluent/plugin/kubernetes_metadata_watch_pods.rb

```diff
@@ -24,9 +24,24 @@ module WatchPods
     include ::KubernetesMetadata::Common
 
     def start_pod_watch
+      if ENV['K8S_NODE_NAME']
+        field_selector = 'spec.nodeName=' + ENV['K8S_NODE_NAME']
+      end
       begin
-        resource_version = @client.get_pods.resourceVersion
-        watcher          = @client.watch_pods(resource_version)
+        options = {
+          resource_version: '0'  # Fetch from API server.
+        }
+        if field_selector
+          options[:field_selector] = field_selector
```

Comment:
if this is the only place `field_selector` is used, then change this block to
```ruby
if ENV['K8S_NODE_NAME']
    options[:field_selector] = 'spec.nodeName=' + ENV['K8S_NODE_NAME']
fi
```

Reply:
Makes sense, thanks!